temporian/implementation/numpy/operators/select_index_values.py

```diff
@@ -35,10 +36,26 @@ def __call__(self, input: EventSet) -> Dict[str, EventSet]:
         output_schema = self.output_schema("output")
 
         keys = self.operator.keys
+        number = self.operator.number
+        fraction = self.operator.fraction
+
+        all_keys = list(input.data.keys())
 
         # Create output EventSet
         output_evset = EventSet(data={}, schema=output_schema)
 
+        if number is not None:
+            if number > len(all_keys):
+                number = len(all_keys)
+            keys = random.sample(all_keys, k=number)
+
+        elif fraction is not None:
+            keys = random.sample(all_keys, k=int(len(all_keys) * fraction))
```

Comment:
Round or ceil?

Reply:
rounds, added note in code and docstring